### PR TITLE
Medium lights and other stuff

### LINKED ISSUE #765: Lights in Containers Show Inappropriately
[BUG]

**Describe the bug**
When you put a light-giving object into a container such as a backpack, when you open the container, the light it makes starts at the location of the object in the inventory menu.

**To Reproduce**
1. Turn on your PDA light.
2. Put your PDA in your backpack.
3. Go somewhere where the bottom-left of the screen is visible but super dark.

**Expected behavior**
Lights should not come from objects in containers, or the light should originate from the actual tile where the container is.

**Screenshots**
image

**Additional context**
I was scared
checked if there was a drone following me
there was not

### PR DESCRIPTION
**Pull request by pali6**
## About the PR 
Alright, there's a lot of internal stuff changed here:
- Reshuffled planes a bit, all "game" planes are now below 0 *including* PLANE_DEFAULT. Mobs now have sight flag SEE_BLACKNESS turned on which displays a black overlay over all tiles you can't see in plane 0. The result of this is that you generally can't see things outside your vision.
- PLANE_UNDERFLOOR is new (without a plane master for now), contains floorcluwnes.
- PLANE_ABOVE_LIGHTING is now called PLANE_OVERLAY_EFFECTS (currently only used for some drugs overlays).
- Made simple lights be visible even if you don't see their source object. So if you drop a PDA behind a corner you'll see the light shining even if you don't see the PDA.
- Made simple lights use objs + vis_contents instead of images. This fixes some visual bugs with the previous point and also makes show_simple_lights, hide_simple_lights not iterate over every single client.
- Added medium lights. Those are basically simple lights of range 2 but they adjust their shape dynamically to match the walls around the object they are attached to.
- Changed the simple light overlay sprite to be a smoother gradient.
- Added some procs that handle both simple and medium lights at once (there's a TODO there to make a smart system to decide whether to use simple or medium lights).
- Also added a sprite for posible future extension of medium lights into directional lights.
### And finally the one actual gameplay change:
- PDA lights have now higher range (and slightly lower brightness), this is accomplished via medium lights.
Before:
dreamseeker_00iGtefBog
After:
dreamseeker_ZZFW1BUS0I
### Actually wait, I also fixed three bugs:
- PDA lights no longer stay online while in your backpack (originating from their screen position). Fixes #765
- Removing a PDA flashlight module no longer makes you keep the light eternally
- PDA flashlights now again display a small lens flare overlay on the PDA's sprite when turned on
- Oh, I think this also fixes some plane mismatches in escape shuttles. I don't quite remember, this PR is a blur

## Why's this needed? 
Simple lights are very nice when it comes to gliding and good for performance. Sadly they are limited to range 1. This should alleviate some of those problems by making the medium lights similar in many respect but with range 2.

## Changelog
```
(u)pali
(*)PDA lights now have higher range (and are slightly less bright).
```

### CODE REVIEW

File: code/SimpleLight.dm [hunk 1]
```diff
@@ -1,8 +1,21 @@
...
+	appearance_flags = RESET_COLOR | RESET_TRANSFORM | RESET_ALPHA | NO_CLIENT_COLOR | KEEP_APART
+	pixel_x = -32
+	pixel_y = -32
+
+/atom
 	var
 		list/simple_light_rgbas = null
 
-		mutable_appearance/simple_light = null
+		obj/overlay/simple_light/simple_light = null
```
Comment (ZeWaka) [inline code/SimpleLight.dm:18]: Can you change these to be in the form `/atom/var/foo/bar = baz`, SpacemanDMM is liable to pick this up as a definition of atom, which is annoying for going to the definition.

File: code/SimpleLight.dm [hunk 2]
```diff
@@ -92,3 +91,155 @@ atom
...
+			if(SOUTH)
+				pixel_y -= 32
+			if(EAST)
+				pixel_x += 32
+			if(WEST)
+				pixel_x -= 32
+			if(0)
+				icon_state = "medium_center"
+
+/atom
```
Comment (ZeWaka) [inline code/SimpleLight.dm:113]: Re: previous comment, could you pretty please absolute path all of these?

File: code/SimpleLight.dm [hunk 3]
```diff
@@ -92,3 +91,155 @@ atom
...
+				light.invisibility = 0
+				continue
+			var/turf/T = get_step(get_turf(src), light.dir)
+			if(T.opacity || T.opaque_atom_count)
+				light.invisibility = 102
+			else
+				light.invisibility = 0
+
+/atom/proc/add_sm_light(id, list/rgba, medium=null)
+	// TODO decide cleverly whether medium or simple if medium == null
```
Comment (ZeWaka) [inline code/SimpleLight.dm:226]: TODO

File: code/atom.dm [hunk 4]
```diff
@@ -3,6 +3,9 @@
 #define DESERIALIZE_NEED_POSTPROCESS 2
 #define DESERIALIZE_NOT_IMPLEMENTED 4
 
+/atom
+	plane = PLANE_DEFAULT
+
```
Comment (ZeWaka) [inline code/atom.dm:8]:
```suggestion
/atom/plane = PLANE_DEFAULT
```

Comment (pali6) [reply code/atom.dm:8]: Why not make atom.dm contain actual definition of /atom?

File: code/SimpleLight.dm (hunk 3 above)
Comment (pali6) [reply code/SimpleLight.dm]: Alright, replaced with a placeholder that probably doesn't work correctly.

File: code/atom.dm (hunk 4 above)
Comment (ZeWaka) [reply code/atom.dm:8]: I think it's defined later in the file already??

Comment (pali6) [reply code/atom.dm:8]: Oh no, I must have missed that somehow, I'll move it.

Comment (pali6) [reply code/atom.dm]: Done.

**Pull request merged**